### PR TITLE
implemented suggestions from code-inspector

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

Implements a few suggestions for code-inspector. Nothing major just a few cases where we didn't add the attribute to the init, a few unnecessary else statements and other minor improvements.

As a result of #593 the code-inspector score increase from 58 (red) to 63 (orange) and I am hoping this PR might bring the score up perhaps a little bit more.

Screenshot from 2020-11-29 23-18-40
Screenshot from 2020-11-25 10-49-26

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: paramak/parametric_reactors/ball_reactor.py [hunk 1]
```diff
@@ -154,6 +152,9 @@ def __init__(
             self.plasma_gap_vertical_thickness,
             self.major_radius - self.minor_radius]
 
+        # calcualted internally by the class
+        self._plasma = None
```
Comment (Shimwell) [inline paramak/parametric_reactors/ball_reactor.py:156]: This was a suggestion of code-inspector but it appears to have broken the tests. Any ideas appreachiated

Comment (RemDelaporteMathurin) [reply paramak/parametric_reactors/ball_reactor.py:156]:
```suggestion
        self.plasma = None
```

Comment (RemDelaporteMathurin) [reply paramak/parametric_reactors/ball_reactor.py:156]: Having _ before attributes hides them for outside the instance. Except if these are defined in __init__. Therefore, initialising self._plasma here will enable the access of the attribute from outside the instance and make the test fail (the test checks that all the attributes with _ are hidden)

Comment (Shimwell) [reply paramak/parametric_reactors/ball_reactor.py:156]: Good to know, wondering if we should rename _plasma to plasma then?

Comment (RemDelaporteMathurin) [reply paramak/parametric_reactors/ball_reactor.py:156]: Up to you. It depends on the reason it was hidden in the first place. If hidden variables are required, I'd suggest to remove all underscores for simplicity

File: paramak/parametric_reactors/center_column_study_reactor.py [hunk 2]
```diff
@@ -98,6 +98,9 @@ def __init__(
             (outer_equatorial_point + inner_equatorial_point) / 2
         self.minor_radius = self.major_radius - inner_equatorial_point
 
+        # calcualted internally by the class
```
Comment (Shimwell) [inline paramak/parametric_reactors/center_column_study_reactor.py:101]:
```suggestion
```

File: paramak/parametric_reactors/center_column_study_reactor.py [hunk 3]
```diff
@@ -98,6 +98,9 @@ def __init__(
...
+        self._plasma = None
```
Comment (Shimwell) [inline paramak/parametric_reactors/center_column_study_reactor.py:102]:
```suggestion
```

File: paramak/parametric_reactors/center_column_study_reactor.py [hunk 4]
```diff
@@ -98,6 +98,9 @@ def __init__(
...
+
```
Comment (Shimwell) [inline paramak/parametric_reactors/center_column_study_reactor.py:103]:
```suggestion
```

File: paramak/parametric_reactors/ball_reactor.py [hunk 5]
```diff
@@ -154,6 +152,9 @@ def __init__(
             self.plasma_gap_vertical_thickness,
             self.major_radius - self.minor_radius]
 
+        # calcualted internally by the class
```
Comment (Shimwell) [inline paramak/parametric_reactors/ball_reactor.py:155]:
```suggestion
```

File: paramak/parametric_reactors/ball_reactor.py (hunk 1 above)
Comment (Shimwell) [inline paramak/parametric_reactors/ball_reactor.py:156]:
```suggestion
```

**Pull request merged**